crypto/s2n_pkey_evp.c

```diff
@@ -276,3 +267,90 @@ int s2n_evp_verify(const struct s2n_pkey *pub, s2n_signature_algorithm sig_alg,
 
     return S2N_SUCCESS;
 }
+
+S2N_RESULT s2n_pkey_evp_size(const struct s2n_pkey *pkey, uint32_t *size_out)
+{
+    RESULT_ENSURE_REF(pkey);
+    RESULT_ENSURE_REF(pkey->pkey);
+
+    const int size = EVP_PKEY_size(pkey->pkey);
+    RESULT_ENSURE_GT(size, 0);
+    *size_out = size;
+
+    return S2N_RESULT_OK;
+}
+
+int s2n_pkey_evp_encrypt(const struct s2n_pkey *key, struct s2n_blob *in, struct s2n_blob *out)
```

Comment:
Are you going to change these to return S2N_RESULT in this refactor?

Reply:
I wasn't planning to. That would change the s2n_pkey interface, so would involve touching even more code (everywhere s2n_pkey is used). I suppose I could do it as a separate follow-up though.

Reply:
I wasn't planning to. I suppose I could though. Probably in a separate PR though-- the cleanup one is already pretty messy.